week4/prime_number.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# How Many Prime Numbers Are in a Given Array
+class PrimeNumber
+  class AttributeError < StandardError; end
+
+  def initialize
+    @array = []
+  end
+
+  def call
+    fill_array
+    count_prime_numbers
+    sum_prime_numbers
+  end
+
+  private
+
+  def fill_array
+    loop do
+      puts "\n When you end filling array 'stop'"
+      print 'Enter Number :  '
+      number = gets.chomp
+      break if number == 'stop'
+
+      validate(number)
+      print @array << number.to_i
+    rescue AttributeError => e
+      puts e
+    end
+  end
+
+  def count_prime_numbers
+    counted = @array.count { |number| is_prime?(number) }
+    puts "there are #{counted} prime numbers"
+  end
+
+  def sum_prime_numbers
+    summarized_prime = finde_prime_number.sum
```

Comment:
also possible `.reduce(:+)`

`summarized_primes` is an appropriate name for this variable